pkg/job/discovery.go

```diff
@@ -98,15 +103,34 @@ func runDiscoveryJob(
 		if data == nil {
 			continue
 		}
+		previousIdx := -1
+		previousID := ""
 		for _, metricDataResult := range data {
 			idx := findGetMetricDataByID(getMetricDatas, *metricDataResult.ID)
+			// TODO: This logic needs to be guarded by a feature flag! Also, remember to add compatibility in the client v2
 			if idx == -1 {
-				logger.Warn("GetMetricData returned unknown metric ID", "metric_id", *metricDataResult.ID)
+				if addHistoricalMetrics {
+					// Use the previousIdx to make a copy
+					if previousIdx != -1 && previousID == *metricDataResult.ID {
+						// Create a new CloudwatchData object
+						newData := *getMetricDatas[previousIdx]
+						newData.GetMetricDataPoint = metricDataResult.Datapoint
+						newData.GetMetricDataTimestamps = metricDataResult.Timestamp
+
+						getMetricDatas = append(getMetricDatas, &newData)
+					} else {
+						logger.Warn("GetMetricData returned unknown metric ID", "metric_id", *metricDataResult.ID)
+					}
+				} else {
+					logger.Warn("GetMetricData returned unknown metric ID", "metric_id", *metricDataResult.ID)
+				}
 				continue
 			}
 			getMetricDatas[idx].GetMetricDataPoint = metricDataResult.Datapoint
 			getMetricDatas[idx].GetMetricDataTimestamps = metricDataResult.Timestamp
 			getMetricDatas[idx].MetricID = nil // mark as processed
+			previousIdx = idx
+			previousID = *metricDataResult.ID
```

Comment:
This is the first attempt but I'm not keen on using the "order" in which the data is retrieved to achieve this. I'm planning to go over this section again and use a hashmap/dictionary to make this logic independent from the order the getMetricDatas has (I believe timestamp descending if I recall correctly)